overlord/snapstate/backend/snapdata_test.go

```diff
@@ -67,7 +68,44 @@ func (s *snapdataSuite) TestRemoveSnapData(c *C) {
 	c.Assert(osutil.FileExists(filepath.Dir(varData)), Equals, true)
 }
 
+func (s *snapdataSuite) TestSnapDataDirs(c *C) {
+	homeDir1 := filepath.Join(s.tempdir, "home", "users")
+	homeDir2 := filepath.Join(s.tempdir, "remote", "users")
+	homeDirs := homeDir1 + "," + homeDir2
+	dirs.SetSnapHomeDirs(homeDirs)
+	dataHomeDirs := []string{filepath.Join(homeDir1, "user1", "snap", "hello", "10"), filepath.Join(homeDir1, "user2", "snap", "hello", "10"),
+		filepath.Join(homeDir2, "user3", "snap", "hello", "10"), filepath.Join(homeDir2, "user4", "snap", "hello", "10")}
+	for _, path := range dataHomeDirs {
+		err := os.MkdirAll(path, 0755)
+		c.Assert(err, IsNil)
+	}
+
+	info := snaptest.MockSnap(c, helloYaml1, &snap.SideInfo{Revision: snap.R(10)})
+	snapDataDirs, err := backend.MockSnapDataDirs(info, nil)
+	c.Assert(err, IsNil)
+	c.Check(snapDataDirs, DeepEquals, dataHomeDirs)
+}
+
+func (s *snapdataSuite) TestSnapCommonDataDirs(c *C) {
+	homeDir1 := filepath.Join(s.tempdir, "home", "users")
+	homeDir2 := filepath.Join(s.tempdir, "remote", "users")
+	homeDirs := homeDir1 + "," + homeDir2
+	dirs.SetSnapHomeDirs(homeDirs)
+	dataHomeDirs := []string{filepath.Join(homeDir1, "user1", "snap", "hello", "common"), filepath.Join(homeDir1, "user2", "snap", "hello", "common"),
+		filepath.Join(homeDir2, "user3", "snap", "hello", "common"), filepath.Join(homeDir2, "user4", "snap", "hello", "common")}
+	for _, path := range dataHomeDirs {
+		err := os.MkdirAll(path, 0755)
+		c.Assert(err, IsNil)
+	}
```

Comment:
I don't understand what this tests are testing as they call Mock functions that are actually not mocking something?